src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1922,12 +1923,89 @@ static bool needsTupleElementNamesAttribute(TypeSymbol type)
                 }
 
                 var resultType = type.VisitType(
-                    predicate: (t, a, b) => !t.TupleElementNames.IsDefaultOrEmpty && !t.IsErrorType(),
+                    predicate: (t, _, _, _) => !t.TupleElementNames.IsDefaultOrEmpty && !t.IsErrorType(),
                     arg: (object?)null);
                 return resultType is object;
             }
         }
 
+        private void CheckUnboundedExtensionErasure(BindingDiagnosticBag diagnostics)
+        {
+            if (this.GetExtendedTypeNoUseSiteDiagnostics(null) is { } extendedType
+                && foundUnboundedErasure(extendedType, extensionsBeingErased: ConsList<TypeSymbol>.Empty.Prepend(this)))
+            {
+                diagnostics.Add(ErrorCode.ERR_CircularBase, Locations[0], extendedType, this);
+            }
+            return;
+
+            static bool foundUnboundedErasure(TypeSymbol type, ConsList<TypeSymbol> extensionsBeingErased, bool isContainer = false)
+            {
+                if (type is NamedTypeSymbol)
+                {
+                    if (!isContainer && type.IsExtension)
+                    {
+                        if (extensionsBeingErased.Contains(type.OriginalDefinition))
+                        {
+                            return true;
+                        }
+
+                        if (type.OriginalDefinition.GetExtendedTypeNoUseSiteDiagnostics(null) is not { } extendedType)
+                        {
+                            return true;
+                        }
+
+                        var newExtensionsBeingErased = extensionsBeingErased.Prepend(type.OriginalDefinition);
+                        if (foundUnboundedErasure(extendedType, newExtensionsBeingErased))
+                        {
+                            return true;
+                        }
+
+                        type = type.GetExtendedTypeNoUseSiteDiagnostics(null)!;
```

Comment:
>type [](http://example.com/codeflow?start=24&length=4)

It looks like we are missing a call to ```foundUnboundedErasure``` for this type when it is not a named type.